packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -66,8 +66,12 @@ function NavigationMenuSelector( {
 
 	const menuChoices = useMemo( () => {
 		return (
-			navigationMenus?.map( ( { id, title } ) => {
-				const label = decodeEntities( title.rendered );
+			navigationMenus?.map( ( { id, title }, index ) => {
+				const label =
+					decodeEntities( title.rendered ) ||
+					/* translators: %s is the index of the menu in the list of menus. */
+					sprintf( __( 'Untitled menu %s' ), index + 1 );
```

Comment:
For posts we use the name "(no title)" in this scenario. Should we do the same here?